src/lib/io/include/sol-uart.h

```diff
@@ -53,36 +53,47 @@ extern "C" {
 
 struct sol_uart;
 
-struct sol_uart *sol_uart_open(const char *port_name);
+enum sol_uart_speed {
+    SOL_UART_SPEED_9600 = 0,
+    SOL_UART_SPEED_19200,
+    SOL_UART_SPEED_38400,
+    SOL_UART_SPEED_57600,
+    SOL_UART_SPEED_115200
+};
+
+enum sol_uart_data_bits {
+    SOL_UART_DATA_BITS_8 = 0,
+    SOL_UART_DATA_BITS_7,
+    SOL_UART_DATA_BITS_6,
+    SOL_UART_DATA_BITS_5
+};
+
+enum sol_uart_parity {
+  SOL_UART_PARITY_DISABLE = 0,
+  SOL_UART_PARITY_ENABLE_EVEN,
+  SOL_UART_PARITY_ENABLE_ODD
+};
+
+enum sol_uart_stop_bits {
```

Comment:
I still think that merging these three enums would be a good idea. Something along these lines:

    enum settings {
           DATA_BITS_8 = 1<<0,
           DATA_BITS_7 = 1<<1,
           DATA_BITS_6 = 1<<2,
           DATA_BITS_5 = 1<<3,
           PARITY_NONE = 1<<4,
           PARITY_EVEN = 1<<5,
           PARITY_ODD = 1<<6,
           STOP_BITS_1 = 1<<7,
           STOP_BITS_2 = 1<<8,
           FLOW_CONTROL_XON = 1<<9,
           FLOW_CONTROL_XOFF = 1<<10,
           DEFAULT = DATA_BITS_8 | PARITY_NONE | STOP_BITS_1
    }

Then it's the matter of:

    uart = uart_open("/dev/ttyS0", BAUD_115200, DEFAULT , cb, data);